xarray/plot/utils.py

```diff
@@ -675,47 +677,48 @@ def _valid_numpy_subdtype(x, numpy_types):
     return any(np.issubdtype(x.dtype, t) for t in numpy_types)
 
 
-def _ensure_plottable(*args):
+def _ensure_plottable(*args) -> None:
     """
     Raise exception if there is anything in args that can't be plotted on an
     axis by matplotlib.
     """
-    numpy_types = [
+    numpy_types: tuple[type[object], ...] = (
         np.floating,
         np.integer,
         np.timedelta64,
         np.datetime64,
         np.bool_,
         np.str_,
-    ]
-    other_types = [datetime]
+    )
+    other_types: tuple[type[object], ...] = (datetime,)
     if cftime is not None:
-        cftime_datetime_types = [cftime.datetime]
-        other_types = other_types + cftime_datetime_types
+        cftime_datetime_types = (cftime.datetime,)
+        other_types += cftime_datetime_types
     else:
-        cftime_datetime_types = []
+        cftime_datetime_types = ()
     for x in args:
         if not (
-            _valid_numpy_subdtype(np.array(x), numpy_types)
-            or _valid_other_type(np.array(x), other_types)
+            _valid_numpy_subdtype(np.asarray(x), numpy_types)
+            or _valid_other_type(np.asarray(x), other_types)
         ):
             raise TypeError(
                 "Plotting requires coordinates to be numeric, boolean, "
                 "or dates of type numpy.datetime64, "
                 "datetime.datetime, cftime.datetime or "
-                f"pandas.Interval. Received data of type {np.array(x).dtype} instead."
-            )
-        if (
-            _valid_other_type(np.array(x), cftime_datetime_types)
-            and not nc_time_axis_available
-        ):
-            raise ImportError(
-                "Plotting of arrays of cftime.datetime "
-                "objects or arrays indexed by "
-                "cftime.datetime objects requires the "
-                "optional `nc-time-axis` (v1.2.0 or later) "
-                "package."
+                f"pandas.Interval. Received data of type {np.asarray(x).dtype} instead."
             )
+        if _valid_other_type(np.asarray(x), cftime_datetime_types):
+            if nc_time_axis_available:
+                # needs to be imported or it will raise an error
+                import nc_time_axis  # noqa: F401
```

Comment:
```suggestion
                # Register cftime datetypes to matplotlib.units.registry, 
                # otherwise matplotlib will raise an error:
                import nc_time_axis  # noqa: F401
```

nc_time_axis adds a bunch of units to matplotlibs registry:
https://github.com/SciTools/nc-time-axis/blob/8f75c7024ad7d7b5f167b991fcbc0128e15dd9ec/nc_time_axis/__init__.py#L620-L622